providers/trello.js

```diff
@@ -0,0 +1,943 @@
+// trello.js
+// https://developers.trello.com/apis/webhooks
+// ========
+const BaseProvider = require('../util/BaseProvider');
+const rpn = require('request-promise-native');
+const urlMod = require('url');
+
+class Trello extends BaseProvider {
+
+    constructor(){
+        super();
+        this.baseLink = 'https://trello.com/';
+        this.avatarurl = 'https://trello-avatars.s3.amazonaws.com/';
+        this.defTrelloColors = {
+            blue: 0x0079bf,
+            yellow: 0xd9b51c,
+            orange: 0xd29034,
+            green: 0x519839,
+            red: 0xb04632,
+            purple: 0x89609e,
+            pink: 0xcd5a91,
+            lime: 0x4bbf6b,
+            sky: 0x00aecc,
+            grey: 0x838c91,
+            trello: 0x026aa7,
+            nocolor: 0xb6bbbf
+        };
+    }
+
+    static getName() {
+        return 'Trello';
+    }
+
+    async getType(){
+        return this.req.body.action.type;
+    }
+
+    // Utility Functions
+
+    static _addMemberThumbnail(avatarHash, embed){
+        if(avatarHash != null && avatarHash !== 'null'){
+            embed.thumbnail = {
+                url: this.avatarurl + avatarHash + '/170.png'
+            };
+        }
+    }
+
+    static _formatLargeString(str, limit = 256){
+        return (str.length > limit ? str.substring(0, limit-1) + '\u2026' : str);
+    }
+
+    static _formatMarkdownHeader(str){
+        const headerRegex = new RegExp('#{1,3} ?(.*)');
+        let match;
+        while(headerRegex.test(str)){
+            match = headerRegex.exec(str);
+            str = str.replace(headerRegex, '**' + match[1] + '**');
+        }
+        return str;
+    }
+
+    static _formatMarkdownImage(str, embed){
+        const imageRegex = new RegExp('!\\[.*\]\\((.*)\\)');
+        if(imageRegex.test(str)){
+            let match = imageRegex.exec(str);
+            embed.image = {url: match[1]};
+            str = str.replace(imageRegex, '');
+        }
+        return str;
+    }
+
+    static _formatMarkdownBullets(str){
+        const bulletRegex = new RegExp('\\*', 'g');
+        if(bulletRegex.test(str)){
+            str = str.replace(bulletRegex, '\u2022');
+        }
+        return str;
+    }
+
+    static _formatMarkdown(str, embed){
+        str = Trello._formatMarkdownBullets(str);
+        str = Trello._formatMarkdownHeader(str);
+        str = Trello._formatMarkdownImage(str, embed);
+
+        return str;
+    }
```

Comment:
The _formatMarkdown function converts the most common/basic markdown formatting into discord-friendly format. Markdown is quite extensive therefore there are formatting types I have not handles. Trello supports markdown on comments, descriptions, etc therefore it's useful to convert some of the formatting to be more readable. Currently, bullet points indicated as asteriks are converted to a bullet character, headers (as indiciated with `#`) are converted to bold (`**`), and one single image will be added to the embed's image field. The full markdown supported by Trello can be found on this link http://help.trello.com/article/821-using-markdown-in-trello . Let me know if I should try to support more of the formatting.